File: helloWorld.js
#!/usr/bin/env node
/**
 * Hello World - Places Data Example
 * A simple JavaScript/Node.js script demonstrating how to work with the places data.
 */

const fs = require('fs');

function parseCSV(csvContent) {
    const lines = csvContent.trim().split('\n');
    const headers = lines[0].split(',');
    const data = [];
    
    for (let i = 1; i < lines.length; i++) {
        const values = lines[i].split(',');
        const obj = {};
        headers.forEach((header, index) => {
            obj[header] = values[index];
        });
        data.push(obj);
    }
    return data;
}

function helloWorld() {
    console.log("Hello World! Welcome to the Places Data Repository");
    console.log("=".repeat(50));
    
    // Read and display CSV data
    console.log("\nPlaces from CSV:");
    try {
        const csvContent = fs.readFileSync('places.csv', 'utf8');
        const places = parseCSV(csvContent);
        places.forEach(place => {
            console.log(`- ${place.name}: ${place.description}`);
            console.log(`  Coordinates: (${place.latitude}, ${place.longitude})`);
        });
    } catch (error) {
        console.log("places.csv not found!");
    }
    
    // Read and display GeoJSON data
    console.log("\nPlaces from GeoJSON:");
    try {
        const geojsonContent = fs.readFileSync('places.geojson', 'utf8');
        const data = JSON.parse(geojsonContent);
        data.features.forEach(feature => {
            const props = feature.properties;
            const coords = feature.geometry.coordinates;
            console.log(`- ${props.name}: ${props.description}`);
            console.log(`  Coordinates: (${coords[1]}, ${coords[0]})`); // lat, lon
        });
    } catch (error) {
        console.log("places.geojson not found!");
    }
}

if (require.main === module) {
    helloWorld();
}

module.exports = { helloWorld };

File: helloWorld.py
#!/usr/bin/env python3
"""
Hello World - Places Data Example
A simple Python script demonstrating how to work with the places data.
"""

import csv
import json

def hello_world():
    """Simple hello world function that displays places data."""
    print("Hello World! Welcome to the Places Data Repository")
    print("=" * 50)
    
    # Read and display CSV data
    print("\nPlaces from CSV:")
    try:
        with open('places.csv', 'r', encoding='utf-8') as file:
            reader = csv.DictReader(file)
            for row in reader:
                print(f"- {row['name']}: {row['description']}")
                print(f"  Coordinates: ({row['latitude']}, {row['longitude']})")
    except FileNotFoundError:
        print("places.csv not found!")
    
    # Read and display GeoJSON data
    print("\nPlaces from GeoJSON:")
    try:
        with open('places.geojson', 'r', encoding='utf-8') as file:
            data = json.load(file)
            for feature in data['features']:
                props = feature['properties']
                coords = feature['geometry']['coordinates']
                print(f"- {props['name']}: {props['description']}")
                print(f"  Coordinates: ({coords[1]}, {coords[0]})")  # lat, lon
    except FileNotFoundError:
        print("places.geojson not found!")

if __name__ == "__main__":
    hello_world()